com.unity.render-pipelines.universal/Editor/Converter/Converters/RenderSettingsConverter.cs

```diff
@@ -0,0 +1,301 @@
+                    Shadows = QualitySettings.shadows,
+                    ShadowResolution = QualitySettings.shadowResolution,
+                    ShadowDistance = QualitySettings.shadowDistance,
+                    ShadowCascadeCount = QualitySettings.shadowCascades,
+                    CascadeSplit2 = QualitySettings.shadowCascade2Split,
+                    CascadeSplit4 = QualitySettings.shadowCascade4Split,
+                    SoftParticles = QualitySettings.softParticles,
+                };
+                m_SettingsItems.Add(projectSettings);
+
+                var setting = QualitySettings.GetRenderPipelineAssetAt(id);
+                var item = new ConverterItemDescriptor {name = $"Quality Level {id}: {levelName}"};
+
+                if (setting != null)
+                {
+                    item.warningMessage = setting.GetType() == typeof(UniversalRenderPipelineAsset)
+                        ? "Contains URP Asset, will override existing asset."
+                        : "Contains SRP Asset, will override existing asset with URP asset.";
+                }
+
+                context.AddAssetToConvert(item);
+                id++;
+            }
+
+            QualitySettings.SetQualityLevel(currentQuality);
+        }
+
+        public override void OnRun(ref RunItemContext context)
+        {
+            var item = context.item;
+            // is quality item
+            if (m_SettingsItems[item.index].GetType() == typeof(RenderSettingItem))
+            {
+                GeneratePipelineAsset(item.index, m_SettingsItems[item.index] as RenderSettingItem);
+            }
+        }
+
+        private void GeneratePipelineAsset(int index, RenderSettingItem settings)
+        {
+            // store current quality level
+            var currentQualityLevel = QualitySettings.GetQualityLevel();
+
+            //creating pipeline asset
+            var asset =
+                ScriptableObject.CreateInstance(typeof(UniversalRenderPipelineAsset)) as UniversalRenderPipelineAsset;
+            if (!AssetDatabase.IsValidFolder($"Assets/{m_PipelineAssetPath}"))
+                AssetDatabase.CreateFolder("Assets", m_PipelineAssetPath);
+            var path = $"Assets/{m_PipelineAssetPath}/{settings.LevelName}_PipelineAsset.asset";
+
+            // Setting Pipeline Asset settings
+            SetPipelineSettings(asset, settings);
+
+            // Create Renderers
+            var defaultIndex = -1;
+            var renderers = new List<ScriptableRendererData>();
+            if (m_RenderingModes.Contains(RenderingMode.Forward))
+            {
+                renderers.Add(CreateRendererDataAsset(path, RenderingPath.Forward, "ForwardRenderer"));
+                if (GetEquivalentRenderMode(m_GraphicsTierSettings.RenderingPath) == RenderingMode.Forward)
+                    defaultIndex = m_RenderingModes.IndexOf(RenderingMode.Forward);
+            }
+
+            if (m_RenderingModes.Contains(RenderingMode.Deferred))
```

Comment:
Should this be an `else if`? Assuming OnInitialize is called once, we are only adding one item to the list ever; 
`m_RenderingModes.Add(GetEquivalentRenderMode(tier.renderingPath));`
If this is the case do we even need to use lists?

Reply:
So this is a little future proofing(for the camera work to be adding in later PR) that the rendering modes will actually contain more than a single value as it will check each camera and add any that differ from using graphics settings.
This is why it's a list atm and also is not a 'else if'.

Reply:
My main worry here is the default index, We need it to get set otherwise we end up with an invalid index. Setting it is behind a few checks which make it brittle.  Also because we check forward first deferred will be preferred as default when both are active?

Reply:
Changed the index to be 0 default, which will make sure at least the first Renderer added to the pipeline asset will be the default one.
Later when the camera stuff is added and there is the chance of having both forward and deferred, forward will alway be added to the pipeline asset first, but depending on what is set in m_GraphicsTierSettings will be used as the default index.